exchange/factory.go

```diff
@@ -29,6 +29,10 @@ import (
 )
 
 // Factory holds the parameters to generate signed exchanges.
+//
+// CertURL can be relative, in which case Factory resolves the absolute URL
+// using the request URL. It should usually contain an absolute path though
```

Comment:
In my opinion, a use of a relative path here would be extremely rare, since a certificate belongs to one (or more) domain(s) (subfolder <--> subdomain doesn't matter here), and the potential confusion could be way bigger than the merit. I'd vote for treating everything (even without leading "/") as an absolute path (to the request host).

Reply:
I agree it's very rare, but I rather prefer to make it the responsibility of the caller. This choice is made for less extra logic (less "おせっかい"); the flexibility is not the primary purpose.

Reply:
Ack. While I agree with making it the caller's responsibility, I don't agree with making it the end user's responsibility (e.g. command-line flags). Fortunately `parseCertURL` in flag_packager.go currently only accepts full URL, adding some comments there would help future readers of this code base, because the rejection is done by checking scheme == https, and now it has more meaning than how the code literally looks like.

Reply:
I just added a TODO.

I'm not sure if I agree with your point about commenting. If I were writing the tool for myself, I'd rather let the tool accept relative paths as-is ― I'd probably get confused if the tool adds the slash automatically. Considering the tool is targeting webmasters rather than Unix geeks, I might introduce a safe guard, i.e. reject paths not starting with `/` or maybe `./`, but I think it's up to who make the change addressing the TODO.

Reply:
Thanks for your thoughts. I respect it.